From 9dbe2549f0180cf1447c7336bebdffec3afd3d73 Mon Sep 17 00:00:00 2001
From: Marek Blaha <mblaha@redhat.com>
Date: Fri, 27 Feb 2026 14:48:01 +0100
Subject: [PATCH] Fix segmentation fault in cmd_requires_privileges

During completion, Session::selected_command is never set. Since
privileges are not required in completion mode, we can return early
before attempting to dereference cmd (which is nullptr).

Issue was introduced in commit 73f51c9e by calling
cmd_requires_privileges() from load_repos().

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=2443105
---
 dnf5/context.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/dnf5/context.cpp b/dnf5/context.cpp
index 26cbc699..7aa44153 100644
--- a/dnf5/context.cpp
+++ b/dnf5/context.cpp
@@ -464,9 +464,14 @@ void Context::Impl::print_error(std::string_view msg) const {
 }
 
 bool Context::Impl::cmd_requires_privileges() const {
-    // the main, dnf5 command, is allowed
     auto cmd = owner.get_selected_command();
+    // During bash completion the selected command in session is not set.
+    // Privileges are not required during completion.
+    if (!cmd) {
+        return false;
+    }
     auto arg_cmd = cmd->get_argument_parser_command();
+    // the main, dnf5 command, is allowed
     if (arg_cmd->get_parent() == nullptr) {
         return false;
     }
-- 
2.53.0

